
/*
      Dopo aver modificato il .proto file:
          python3 -m grpc_tools.protoc -I../../protos --python_out=. --grpc_python_out=. ../../protos/doubleauth.proto

      Questo genera
          - newpassword_pb2.py che contiene le classi di richiesta e risposta
          - newpassword_pb2_grpc.py che contiene le classi client e server

*/

syntax = "proto3";

option java_multiple_files = true;
option java_package = "/SDCC/services";
option java_outer_classname = "DoubleAuth";
option objc_class_prefix = "DA";
package newpw;

// The  service definition.
service Doubleauth {

  rpc doLoginThirdPart (GenCodeRequest) returns (Reply) {}
  rpc checkCode (CheckCodeRequest) returns (Reply) {}
  rpc registrationThirdPart (RegistrationRequest) returns (Reply) {}

}

message GenCodeRequest{
  string email = 1;
  string service = 2;
}

message CheckCodeRequest{
  string email = 1;
  string service = 2;
  string code = 3;
}

message RegistrationRequest{
  string email = 1;
  string service = 2;
}

message Reply{
  bool message = 1;
}
